pystac/extensions/hooks.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABC, abstractmethod
```

Comment:
I also noticed this and was a little annoyed about how the hooks and base were at the same level of the extensions, but couldn't come up with a good solution (though I didn't think about it too hard). One way would be to do the sort of reverse of what you're saying and have all extensions move from `.py` to having their own packages, so that all directories in the `extensions` directories are extensions. This might also make it easier for readability as some of those extension files have a lot of types in a single file, and having a package directory would allow things to be broken out more and more findable. If that seems like something worth doing, or if this should be explored more, perhaps write an issue and/or make a follow up PR?

Reply:
Yeah, I'll write up an issue to follow up on this. It could definitely benefit from more discussion on whether it's worth it and how to best structure those files.